### DISCUSSION TITLE
Read Per-Frame MetaData on TIFF -- Allows CMYK & RGB Detection

### DISCUSSION THREAD

**Discussion by pha3z** [SixLabors/ImageSharp#2286] [Ideas]
I was told to raise an Idea about this.

I wanted to detect CMYK vs RGB data on TIFF, but we need support for per-frame MetaData on TIFF.

Original discussion:
[https://github.com/SixLabors/ImageSharp/discussions/2278](https://github.com/SixLabors/ImageSharp/discussions/2278)

**Comment by IldarKhayrutdinov** [comment]
I could do it, @JimBobSquarePants I guess `Identify` should work really fast, should I do instant  loading of frames or optionally/lazy?

**Reply by JimBobSquarePants** [reply]
Thanks @IldarKhayrutdinov !!

I would say instant loading. The main cost of the operation is pixel decoding and we do not want to hold any part of the input data as a reference

**Reply by IldarKhayrutdinov** [reply]
fixed by https://github.com/SixLabors/ImageSharp/pull/2363
